### PR TITLE
Resolve crash when retrieving diagnostic location

### PR DESCRIPTION
**Pull request by jeremyong**
The implementation of IDxcDiagnostic::GetLocation was missing a local instance of DxcThreadMalloc. Later, in DxcSourceLocation::Create, this could result in a crash in various multithreaded uses due to the TLS allocator being unset.

### CODE REVIEW

File: tools/clang/unittests/HLSL/DXIsenseTest.cpp [hunk 1]
```diff
@@ -237,6 +238,16 @@ TEST_F(DXIntellisenseTest, InclusionWhenMissingThenError) {
   VERIFY_SUCCEEDED(pDiag->GetSeverity(&Severity));
   VERIFY_IS_TRUE(Severity == DxcDiagnosticSeverity::DxcDiagnostic_Error ||
     Severity == DxcDiagnosticSeverity::DxcDiagnostic_Fatal);
+  VERIFY_SUCCEEDED(pDiag->GetLocation(&location));
```
Comment (llvm-beanz) [inline tools/clang/unittests/HLSL/DXIsenseTest.cpp:241]: We should also have a test verifying the failure if the parameter is null.

Comment (jeremyong) [reply tools/clang/unittests/HLSL/DXIsenseTest.cpp:242]: Done

**Review by llvm-beanz — APPROVED** [review]
LGTM

**Comment by damyanp** [comment]
This PR was closed as it has not been updated in the last two years. Please feel free to reopen if this PR should be merged and is in a reviewable state.

**Comment by damyanp** [comment]
This PR was closed as it has not been updated in the last two years. Please feel free to reopen if this PR should be merged and is in a reviewable state.
